### DISCUSSION TITLE
Posts Index

### DISCUSSION THREAD

**Discussion by sistematico** [posva/unplugin-vue-router#331] [Questions and Help]
Anyway to implement a Blog Posts Index like this: https://github.com/antfu/antfu.me/blob/main/src/components/ListPosts.vue using `unplugin-vue-router`?

**Comment by posva** [comment]
What did you try?

**Reply by sistematico** [reply]
First, thank you for this plugin.

Eduardo, I'm little lost at this point, this is my test using `vite-plugin-pages`: 
```
    pages({
      extensions: ['vue', 'md'],
      extendRoute(route, _parent) {
        const path = resolve(__dirname, route.component.slice(1))

        let layout = ''

        if (path.includes('/posts')) layout = 'blog'
        else if (path.includes('/projetos')) layout = 'projects'

        if (!path.includes('projects.md') && path.endsWith('.md')) {
          // const layout = path.includes('/posts') ? 'blog' : ''
          const md = fs.readFileSync(path, 'utf-8')
          const { data } = matter(md)
          route.meta = Object.assign(route.meta || {}, { frontmatter: data })
        }

        route.meta = Object.assign(route.meta || {}, { layout })
        return route
      }
    }),
```

But i can't find filename(to assign frontmatter) using `unplugin-vue-router`.

https://github.com/sistematico/lucasbrum.net/blob/main/src/pages/posts/index.vue

Any tip is welcome, thank you for your response.
